File: ocaml/raindrops/raindrops.ml
let raindrop num = let open Base in [(3, "Pling");(5, "Plang");(7, "Plong")]
  |> List.fold ~f:(fun acc (i, s) -> if num % i = 0 then acc ^ s else acc) ~init:""
  |> function "" -> Int.to_string num | s -> s

File: go/raindrops/raindrops.go
package raindrops

import "strconv"

var content = []struct {
	nb  int
	str string
}{ {3, "Pling"}, {5, "Plang"}, {7, "Plong"} }

func Convert(number int) (acc string) {
	for _, el := range content { if number % el.nb == 0 { acc += el.str } }
	if acc == "" { acc = strconv.Itoa(number) }
	return
}
